### ISSUE TITLE
Encoding problem with ESX task

### ISSUE THREAD

**Issue by 575544685** [glpi-project/glpi-agent#204]
### Describe the bug

I create a ESX task to do an inventory on a ESXi server. Some virrual machines on this server are in Chinese. The task runs well but Chinese characters are unreadable on GLPI server. 
![image](https://user-images.githubusercontent.com/111935528/188402260-5b21cb2e-6511-4bc2-95d3-9451c86e8191.png)
Then I  delete the ESX server created by this ESX task and do a local ESX inventory and then upload the result to the GLPI server by glpi-injector command. The Characters are readable this time. 
![image](https://user-images.githubusercontent.com/111935528/188402995-9f5a9e5b-305c-4d08-9e3f-a70baca85d9e.png)

### To reproduce

1. Create virtual machines on ESI sever with Chinese character names.
2. Set Character set of Mariadb(mysql) to utf8/utf8mb4.
3. Create an ESX task on GLPI server and run.
4. Check the names of virtual machines of the ESX server. 

### Expected behavior

The virutal machines' names are readable.

### Operating system

Linux

### GLPI Agent version

1.4

### GLPI version

10.0.2

### GLPIInventory plugin

1.0.4

**Comment by trasher** [comment]
please provide example inventory file

**Comment by g-bougard** [comment]
Hi @575544685 
you can create such file with `glpi-esx` script.

**Comment by 575544685** [comment]
Here is the local inventory file,.(I change the suffix to txt to upload it). The characters are readable when I uploaded it by injector or imported it from GLPI server. 
[localhost-2022-09-05-14-49-58.ocs.txt](https://github.com/glpi-project/glpi-inventory-plugin/files/9488323/localhost-2022-09-05-14-49-58.ocs.txt)

Here is the inventory file I downloaded from GLPI server.(The virtual machines' name are unreadable):
[computer_0_33.json.txt](https://github.com/glpi-project/glpi-inventory-plugin/files/9488390/computer_0_33.json.txt)

**Comment by trasher** [comment]
Seems like it's ok importing your file into GLPI:
![image](https://user-images.githubusercontent.com/224733/188413370-1370fb41-632c-4cdf-a88d-e3791cbd5537.png)

Maybe a config issue on you glpi web server's config, or in the agent; I have no idea

**Comment by 575544685** [comment]
Here is another inventory file I downloaded from GLPI server.(The ESX server is generated by injector command. The virtual machines' names are readable)
[computer_0_35.xm.txt](https://github.com/glpi-project/glpi-inventory-plugin/files/9488558/computer_0_35.xmHere is another inventory file I downloaded from GLPI server.(The ESX server is generated by injector command. The virtual machines' names are readable)
[computer_0_35.xm.txt](https://github.com/glpi-project/glpi-inventory-plugin/files/9488558/computer_0_35.xm.txt).txt)

**Comment by 575544685** [comment]
Here is another inventory file I downloaded from GLPI server.(The ESX server is generated by injector command. The virtual machines' names are readable)
[computer_0_35.xm.txt](https://github.com/glpi-project/glpi-inventory-plugin/files/9488704/computer_0_35.xm.txt)

**Comment by 575544685** [comment]
@trasher  The  inventory file [localhost-2022-09-05-14-49-58.ocs] is OK. The file is generated by local command line. But if I do a ESX task from GLPI server directly, the characters are unreadable.

**Comment by 575544685** [comment]
You can try to import the file computer_0_33.json.txt. This is the inventory file generated by ESX task on GLPI server.

**Comment by g-bougard** [comment]
Hi @575544685 
for now I think this is an agent issue which probably forget to decode answers from ESX server. This could explain why the glpi-esx script still can output a proper XML while received json on glpi side could be wrong.
Can you just confirm one point: on which OS your glpi-agent is running ?

**Comment by 575544685** [comment]
WIN10PRO

**Comment by 575544685** [comment]
To avoid misunderstanding, I clarify the data flow again: 
1. Glpi-esx --ocs file---->glpi-injector(glpi agent based on Win10) -------->Glpi server (based on CentOS)----download---> xxx.XML---import-->Glpi server
Everything is fine when I do these above..
2. Create and run Esx task from glpi server ---->ubreadable name ------download the inventory file---->xxx.json-----import--->glpi server-----still unreadable

**Comment by g-bougard** [comment]
Hi @575544685 
I just pushed a fix for this issue. You can try the development build which should be available soon from Windows-Build artifact at https://github.com/glpi-project/glpi-agent/actions/runs/2993990766
Or you can just try next nightly build you'll find tomorrow on https://nightly.glpi-project.org/glpi-agent/
Or you can apply yourself the patch: see the last commit and you need to only apply the patch on lib/GLPI/Agent/SOAP/VMware.pm file.

Can you confirm this fixes your issue ?

**Comment by 575544685** [comment]
@g-bougard My agent is based on Win10, where can I find the VMware.pm? Where is the patch file and how can I apply it? Thx.

**Comment by 575544685** [comment]
@g-bougard I downloaded the Windows build and reinstalled the agent. The issuse is solved.

**Comment by 575544685** [comment]
I checked the log file of the new agent. There is an error message:
[Tue Sep  6 13:34:15 2022][info] New inventory from LIANGYONG.thtf.com-2022-08-08-14-57-19 for local0
[Tue Sep  6 13:34:15 2022][error] Can't open HKEY_LOCAL_MACHINE/SOFTWARE/Open Source key: Invaid Handle

I also checked the regiser tabel fo Windows, there is no "Open Source key" indeed. I didn't find this error message when I used Agent 1.4. What is this key for? How can I fix this error?
